include/crm/common/xml_internal.h

```diff
@@ -354,4 +354,25 @@ pcmk__xe_foreach_child(xmlNode *xml, const char *child_element_name,
                        void (*handler)(xmlNode *xml, void *userdata),
                        void *userdata);
 
+/*!
+ * \internal
+ * \brief Iterate over child elements of \p xml, performing some
+ *        callback function on each node.  This function will stop iterating
+ *        on the first element for which \p handler returns \p true.
+ *
+ * \param[in] xml                 The starting XML node
+ * \param[in] child_element_name  The name that the node must match in order
+ *                                for \p handler to be run.  If NULL, all
+ *                                child elements will match.
+ * \param[in] handler             The callback function.
+ * \param[in] userdata            User data to pass to the callback function.
+ *                                Can be NULL.
+ *
+ * \return \p true if \p handler returns true, \p false otherwise.
+ */
+bool
+pcmk__xe_any_child(xmlNode *xml, const char *child_element_name,
+                   bool (*handler)(xmlNode *xml, void *userdata),
+                   void *userdata);
```

Comment:
I like this better actually, this could be the implementation of pcmk__xe_foreach_child(). It's easy enough to have a handler always return false.

It might be even more flexible to return a standard code, stopping if it's not pcmk_rc_ok. Then the handler could get back a little more info to the caller.

Reply:
I think having the handler return a standard code is probably the best approach.  I considered that, but things like `command_delete` in tools/crm_attribute.c have a pretty weird condition on the return code in the loop.  But thinking about it some more, I don't want to build an API around one weird condition.  So I think we can go with the return code and then that one function can just be different.

I also think eventually we're going to want something like:
```
xmlNode *
pcmk__xe_matching_child(xmlNode *xml, const char *child_element_name,
                        bool (*find)(xmlNode *xml, void *userdata),
                        void *userdata);
```

That acts similarly but returns the first child for which `find` returns true.  That sort of thing comes up from time to time in the loops I've looked at.